### PR TITLE
[pretty] update `clang-format` and `clang-tidy` to v14

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: examples/platforms/simulation/trel.c [hunk 1]
```diff
@@ -366,8 +366,8 @@ void otPlatTrelRegisterService(otInstance *aInstance, uint16_t aPort, const uint
 #endif
 }
 
-void otPlatTrelSend(otInstance *      aInstance,
-                    const uint8_t *   aUdpPayload,
+void otPlatTrelSend(otInstance       *aInstance,
+                    const uint8_t    *aUdpPayload,
```
Comment (abtink) [inline examples/platforms/simulation/trel.c:370]: Is there a way to keep `*` same as before (to reduce the `diff`)?

**Review by abtink — APPROVED** [review]

File: examples/platforms/simulation/trel.c (hunk 1 above)
Comment (abtink) [reply examples/platforms/simulation/trel.c:370]: I found `PointerAlignment` and maybe setting it to `Left` can work?

Comment (jwhui) [reply examples/platforms/simulation/trel.c:370]: Setting `PointerAlignment=PAS_Left` would remove the space between type and `*`. For example:
```
void otPlatTrelSend(otInstance*       aInstance,
                    const uint8_t*    aUdpPayload,
```

Comment (abtink) [reply examples/platforms/simulation/trel.c:370]: I guess we end up with a `diff` for all case.  I would say the right alignment `otInstance       *aInstance` looks nicer than `otInstance*` . 

There seems to be a `PAS_Middle` option as well.

Comment (abtink) [reply examples/platforms/simulation/trel.c:370]: With `Middle` do we still get `diff` for the multi-line case (I guess we may if it tries to add space on both sides of `*` or `&`).

I think no matter what we end up with some big `diff`. Looks-wise I like the right alignment `otInstance  *aInstance` over other options. 😃

**Comment by lmnotran** [comment]
Thanks for handling this, @jwhui !

File: examples/platforms/simulation/trel.c (hunk 1 above)
Comment (abtink) [reply examples/platforms/simulation/trel.c:370]: Thanks. `Right` has the smallest `diff` and looks better than others (in my view) :)

**Pull request merged**